pkg/controllers/teamrbac/teamrolebinding_controller.go

```diff
@@ -99,50 +101,70 @@ func (r *TeamRoleBindingReconciler) Reconcile(ctx context.Context, req ctrl.Requ
+					r.recorder.Eventf(teamRoleBinding, corev1.EventTypeWarning, greenhousev1alpha1.FailedDeleteRoleBindingReason, "Error retrieving RoleBinding %s for cluster %s in namespace %s", teamRoleBinding.GetRBACName(), cluster.GetName(), namespace)
 					// TODO: collect errors and return them
 					return ctrl.Result{}, err
 				default:
 					if err := remoteRestClient.Delete(ctx, remoteObject); err != nil {
-						r.recorder.Eventf(teamRoleBinding, corev1.EventTypeWarning, greenhousev1alpha1.FailedDeleteRoleBindingReason, "Error deleting RoleBinding %s for cluster %s in namespace %s", teamRoleBinding.GetName(), cluster.GetName(), namespace)
+						r.recorder.Eventf(teamRoleBinding, corev1.EventTypeWarning, greenhousev1alpha1.FailedDeleteRoleBindingReason, "Error deleting RoleBinding %s for cluster %s in namespace %s", teamRoleBinding.GetRBACName(), cluster.GetName(), namespace)
 						// TODO: collect errors and return them
 						return ctrl.Result{}, err
 					}
 				}
 			}
-			deletionSuccessful = true
+			rbacBindingsDeleted = true
 		case false:
-			remoteObjectKey := types.NamespacedName{Name: teamRoleBinding.GetName()}
+			remoteObjectKey := types.NamespacedName{Name: teamRoleBinding.GetRBACName()}
 			remoteObject := &rbacv1.ClusterRoleBinding{}
 			err := remoteRestClient.Get(ctx, remoteObjectKey, remoteObject)
-			switch {
-			case apierrors.IsNotFound(err):
-				deletionSuccessful = true
-			case err != nil:
-				r.recorder.Eventf(teamRoleBinding, corev1.EventTypeWarning, greenhousev1alpha1.FailedDeleteRoleBindingReason, "Error retrieving ClusterRoleBinding %s for cluster %s", teamRoleBinding.GetName, cluster.GetName())
+			if err != nil && !apierrors.IsNotFound(err) {
+				r.recorder.Eventf(teamRoleBinding, corev1.EventTypeWarning, greenhousev1alpha1.FailedDeleteRoleBindingReason, "Error retrieving ClusterRoleBinding %s for cluster %s", teamRoleBinding.GetRBACName(), cluster.GetName())
 				// TODO: collect errors and return them
-			default:
-				if err := remoteRestClient.Delete(ctx, remoteObject); err != nil {
-					r.recorder.Eventf(teamRoleBinding, corev1.EventTypeWarning, greenhousev1alpha1.FailedDeleteClusterRoleBindingReason, "Error deleting ClusterRoleBinding %s for cluster %s", teamRoleBinding.GetName, cluster.GetName())
-					// TODO: collect errors and return them
-					return ctrl.Result{}, err
-				} else {
-					deletionSuccessful = true
-				}
+				return ctrl.Result{}, err
 			}
+			if err := remoteRestClient.Delete(ctx, remoteObject); err != nil {
+				r.recorder.Eventf(teamRoleBinding, corev1.EventTypeWarning, greenhousev1alpha1.FailedDeleteClusterRoleBindingReason, "Error deleting ClusterRoleBinding %s for cluster %s", teamRoleBinding.GetRBACName(), cluster.GetName())
+				// TODO: collect errors and return them
```

Comment:
```suggestion
			
```

see above

Reply:
same as above. this will get relevant again, once the `clusterSelector` returns with https://github.com/cloudoperators/greenhouse/issues/90. I'd keep it in to not forget